### ISSUE TITLE
file creation: triggering files

### ISSUE THREAD

**Issue by oliver-sanders** [metomi/rose#2300]
A user has noticed that it is possible to use metadata triggers to trigger file creation. This may be an accidental feature, it isn't documented, nor is it covered in the test-battery. It makes logical sense to me that we should be able to do this, it provides a nice clean way to avoid dumping out unneeded files.

@metomi/core Should this become a supported feature?

#### Example:

*rose-app.conf*
```ini
[command]
default=true

[file:foo]
source=namelist:foo

[namelist:foo]
bar=bart
foo=foot
switch=.false.
```

*meta/rose-meta.conf*
```ini
[file:foo]

[namelist:foo=switch]
type=logical
trigger=file:foo: .true.
```

**Comment by dpmatthews** [comment]
Makes sense to me

**Comment by oliver-sanders** [comment]
We should:

* Add a test to ensure this feature is properly supported.
* Document it [here](http://metomi.github.io/rose/doc/html/api/configuration/metadata.html?highlight=trigger#rose:conf:rose-meta.conf[SETTING]trigger) and perhaps link to that from [here](http://metomi.github.io/rose/doc/html/api/configuration/file-creation.html)

**Comment by sadielbartholomew** [comment]
Addressed fully by #2389.
